sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/OnHeapColumnVector.java

```diff
@@ -366,55 +362,23 @@ public double getDouble(int rowId) {
     }
   }
 
-  //
-  // APIs dealing with Arrays
-  //
-
-  @Override
-  public int getArrayLength(int rowId) {
-    return arrayLengths[rowId];
-  }
-  @Override
-  public int getArrayOffset(int rowId) {
-    return arrayOffsets[rowId];
-  }
-
-  @Override
-  public void putArray(int rowId, int offset, int length) {
-    arrayOffsets[rowId] = offset;
-    arrayLengths[rowId] = length;
-  }
-
   @Override
   public void loadBytes(ColumnVector.Array array) {
     array.byteArray = byteData;
     array.byteArrayOffset = array.offset;
   }
 
-  //
-  // APIs dealing with Byte Arrays
-  //
-
-  @Override
-  public int putByteArray(int rowId, byte[] value, int offset, int length) {
-    int result = arrayData().appendBytes(length, value, offset);
-    arrayOffsets[rowId] = result;
-    arrayLengths[rowId] = length;
-    return result;
-  }
-
   // Spilt this function out since it is the slow path.
   @Override
   protected void reserveInternal(int newCapacity) {
     if (this.resultArray != null || DecimalType.isByteArrayDecimalType(type)) {
-      int[] newLengths = new int[newCapacity];
-      int[] newOffsets = new int[newCapacity];
-      if (this.arrayLengths != null) {
-        System.arraycopy(this.arrayLengths, 0, newLengths, 0, capacity);
-        System.arraycopy(this.arrayOffsets, 0, newOffsets, 0, capacity);
+      // need 2 ints as offset and length for each array.
```

Comment:
It would be good to add a comment regarding that `intData[]` is used for other purpose compared to the original intention (i.e. storing int column data).